src/main/java/net/dv8tion/jda/internal/JDAImpl.java

```diff
@@ -675,6 +676,53 @@ public SnowflakeCacheView<RichCustomEmoji> getEmojiCache()
         return CacheView.allSnowflakes(() -> guildCache.stream().map(Guild::getEmojiCache));
     }
 
+    @Nonnull
+    @Override
+    public RestAction<ApplicationEmoji> createApplicationEmoji(@Nonnull String name, @Nonnull Icon icon)
+    {
+        Checks.inRange(name, 2, 32, "Emoji name");
+        Checks.notNull(icon, "Emoji icon");
+
+        DataObject body = DataObject.empty();
+        body.put("name", name);
+        body.put("image", icon.getEncoding());
+
+        final Route.CompiledRoute route = Route.Applications.CREATE_APPLICATION_EMOJI.compile(getSelfUser().getApplicationId());
+        return new RestActionImpl<>(this, route, body, (response, request) ->
+        {
+            final DataObject obj = response.getObject();
+            return entityBuilder.createApplicationEmoji(this, obj);
+        });
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<List<ApplicationEmoji>> retrieveApplicationEmojis()
+    {
+        Route.CompiledRoute route = Route.Applications.GET_APPLICATION_EMOJIS.compile(getSelfUser().getApplicationId());
+        return new RestActionImpl<>(this, route, (response, request) ->
+        {
+            DataArray emojis = response.getObject().getArray("items");
+            List<ApplicationEmoji> list = new ArrayList<>(emojis.length());
+            for (int i = 0; i < emojis.length(); i++)
+            {
+                list.add(entityBuilder.createApplicationEmoji(this, emojis.getObject(i)));
+            }
+
+            return Collections.unmodifiableList(list);
+        });
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<ApplicationEmoji> retrieveApplicationEmojiById(@Nonnull String emojiId)
```

Comment:
Add a `Checks.isSnowflake` for the `emojiId`

Reply:
Make sure the `retrieveApplicationEmojiById` and other methods that reference mention the exceptions emitted by `Checks.isSnowflake`